docs/utilities/parser.md

```diff
@@ -2,168 +2,85 @@
 
-This utility provides data parsing and deep validation using [Pydantic](https://pydantic-docs.helpmanual.io/){target="_blank" rel="nofollow"}.
+The Parser utility simplifies data parsing and validation using [Pydantic](https://pydantic-docs.helpmanual.io/){target="\_blank" rel="nofollow"}. It allows you to define data models in pure Python classes, parse and validate incoming events, and extract only the data you need.
 
 ## Key features
 
-* Defines data in pure Python classes, then parse, validate and extract only what you want
-* Built-in envelopes to unwrap, extend, and validate popular event sources payloads
-* Enforces type hints at runtime with user-friendly errors
-* Support only Pydantic v2
+- Define data models using Python classes
+- Parse and validate Lambda event payloads
+- Built-in support for common AWS event sources
+- Runtime type checking with user-friendly error messages
+- Compatible with Pydantic v2
 
 ## Getting started
 
 ### Install
 
-!!! info "This is not necessary if you're installing Powertools for AWS Lambda (Python) via [Lambda Layer/SAR](../index.md#lambda-layer){target="_blank"}"
-
-You need to bring Pydantic v2.4.0 or later as an external dependency.
-
-Add `aws-lambda-powertools[parser]` as a dependency in your preferred tool: _e.g._, _requirements.txt_, _pyproject.toml_.
-
-### Defining models
-
-You can define models to parse incoming events by inheriting from `BaseModel`.
-
-```python title="Defining an Order data model"
-from aws_lambda_powertools.utilities.parser import BaseModel
-from typing import List, Optional
-
-class OrderItem(BaseModel):
-	id: int
-	quantity: int
-	description: str
+Parser supports Pydantic v2. Each Pydantic version requires different dependencies before you can use Parser.
 
-class Order(BaseModel):
-	id: int
-	description: str
-	items: List[OrderItem] # nesting models are supported
-	optional_field: Optional[str] = None # this field may or may not be available when parsing
+```python
+pip install aws-lambda-powertools
 ```
 
-These are simply Python classes that inherit from BaseModel. **Parser** enforces type hints declared in your model at runtime.
+!!! info "This is not necessary if you're installing Powertools for AWS Lambda (Python) via [Lambda Layer/SAR](../index.md#lambda-layer){target="\_blank"}"
+
+???+ warning
+    This will increase the compressed package size by >10MB due to the Pydantic dependency.
 
-### Parsing events
+    To reduce the impact on the package size at the expense of 30%-50% of its performance[Pydantic can also be
+    installed without binary files](https://pydantic-docs.helpmanual.io/install/#performance-vs-package-size-trade-off){target="_blank" rel="nofollow"}:
 
-You can parse inbound events using **event_parser** decorator, or the standalone `parse` function. Both are also able to parse either dictionary or JSON string as an input.
+    Pip example:`SKIP_CYTHON=1 pip install --no-binary pydantic aws-lambda-powertools[parser]`
```

Comment:
You can remove this block. Pydantic v2 doesn't work like that.

Reply:
Removed!